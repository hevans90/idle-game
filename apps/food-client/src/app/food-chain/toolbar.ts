import * as PIXI from 'pixi.js';
import { Player } from './player';
import { createSprite } from './utils/graphics-utils';
import { app } from './utils/singletons';

export const renderToolbar = (player: Player) => {
  const previousToolbar = app.stage.getChildByName('toolbar');
  app.stage.removeChild(previousToolbar);
  const toolbar = new PIXI.Container();

  const graphic = new PIXI.Graphics();
  graphic.lineStyle(2, 0x000000);
  graphic.beginFill(player.colour);
  graphic.drawRect(0, 0, app.screen.width, 50);
  graphic.endFill();
  toolbar.addChild(graphic);

  const cashSprite = createSprite('cash', 40);
  Object.assign(cashSprite, { x: 20, y: 5 });
  const cashText = new PIXI.Text(
    `$${player.cash}`,
    new PIXI.TextStyle({
      fontFamily: 'zx-spectrum',
      fontSize: 24,
      fontWeight: 'bold',
      fill: '#ffffff',
    })
  );
  Object.assign(cashText, { x: 70, y: 15 });
  toolbar.addChild(cashSprite, cashText);

  Object.entries(player.food).forEach((food, i) => {
    const foodKindName = food[0];
    const foodAmount = food[1];
    const foodSprite = createSprite(foodKindName, 40);
    Object.assign(foodSprite, { x: 20 + (i + 1) * 100, y: 5 });

    const foodAmountText = new PIXI.Text(
      `x ${foodAmount.amount}`,
      new PIXI.TextStyle({
        fontFamily: 'zx-spectrum',
        fontSize: 24,
        fontWeight: 'bold',
        fill: '#ffffff',
      })
    );
    player.food[foodKindName].sprite = foodSprite;
    player.food[foodKindName].textSprite = foodAmountText;
    Object.assign(foodAmountText, { x: 60 + (i + 1) * 100, y: 15 });
    toolbar.addChild(foodSprite, foodAmountText);
  });

  const playerNameText = new PIXI.Text(
    player.name,
    new PIXI.TextStyle({
      fontFamily: 'zx-spectrum',
      fontSize: 24,
      fontWeight: 'bold',
      fill: '#ffffff',
    })
  );
  Object.assign(playerNameText, {
    x: app.screen.width - playerNameText.width - 50,
    y: 12,
  });
  toolbar.addChild(playerNameText);

  toolbar.name = 'toolbar';
  app.stage.addChild(toolbar);
};
