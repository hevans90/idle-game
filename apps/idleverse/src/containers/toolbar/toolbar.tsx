import { Box, Button, useColorMode, useColorModeValue } from '@chakra-ui/react';
import { Auth } from '../../_auth/auth';

export const ToolBar = () => {
  const { colorMode, toggleColorMode } = useColorMode();

  const color = useColorModeValue('gray.200', 'gray.700');

  return (
    <Box
      className="toolbar"
      padding={2}
      display="flex"
      alignItems="center"
      justifyContent="space-between"
      bgColor={color}
    >
      <Button onClick={toggleColorMode}>
        Toggle {colorMode === 'light' ? 'Dark' : 'Light'}
      </Button>

      <Auth></Auth>
    </Box>
  );
};
