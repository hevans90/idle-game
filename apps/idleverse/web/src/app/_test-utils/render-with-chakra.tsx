import { ChakraProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import React from 'react';
import { theme } from '../_theme/theme';

export const renderWithChakra = (ui) => {
  const Wrapper = ({ children }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return render(ui, { wrapper: Wrapper });
};
