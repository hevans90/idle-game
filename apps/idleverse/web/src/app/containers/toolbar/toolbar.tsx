import { Box, Kbd, useColorModeValue, useDisclosure } from '@chakra-ui/react';
import { useKeypress } from '../../hooks/use-keypress';
import { Auth } from '../../_auth/auth';
import { EscMenuContainer } from '../esc-menu/escape-menu.container';

export const ToolBar = () => {
  const color = useColorModeValue('gray.200', 'gray.700');

  const borderColor = useColorModeValue('gray.200', 'gray.600');

  useKeypress('Escape', () => isOpen || onOpen());

  const { isOpen, onOpen, onClose } = useDisclosure();

  return (
    <>
      <EscMenuContainer isOpen={isOpen} onClose={onClose}></EscMenuContainer>
      <Box
        className="toolbar"
        padding={2}
        display="flex"
        alignItems="center"
        justifyContent="space-between"
        bgColor={color}
        borderColor={borderColor}
        borderBottomStyle="solid"
        borderBottomWidth="1px"
      >
        <span>
          Settings: <Kbd>Esc</Kbd>
        </span>
        <Auth></Auth>
      </Box>
    </>
  );
};
