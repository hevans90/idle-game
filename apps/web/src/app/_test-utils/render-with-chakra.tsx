import { ChakraProvider } from '@chakra-ui/react';
import { theme } from '@idleverse/theme';
import { render } from '@testing-library/react';

export const renderWithChakra = (ui) => {
  const Wrapper = ({ children }) => (
    <ChakraProvider theme={theme({ primary: 'gray', secondary: 'teal' })}>
      {children}
    </ChakraProvider>
  );

  return render(ui, { wrapper: Wrapper });
};
