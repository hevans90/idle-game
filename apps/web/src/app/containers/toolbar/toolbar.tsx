import { useReactiveVar } from '@apollo/client';
import { ChatIcon, SettingsIcon } from '@chakra-ui/icons';
import {
  Box,
  Button,
  HStack,
  Kbd,
  useColorModeValue,
  useDisclosure,
} from '@chakra-ui/react';
import { useKeypress } from '../../hooks/use-keypress';
import { Auth } from '../../_auth/auth';
import { responsiveFontProps } from '../../_responsive-utils/font-props';
import { layoutVar } from '../../_state/persisted-reactive-variables';
import { EscMenuContainer } from '../esc-menu/escape-menu.container';

export const ToolBar = () => {
  const color = useColorModeValue('gray.200', 'gray.700');

  const borderColor = useColorModeValue('gray.200', 'gray.600');

  useKeypress('Escape', () => isOpen || onOpen());

  const { isOpen, onOpen, onClose } = useDisclosure();

  const { sideNav, toolBar } = useReactiveVar(layoutVar);

  return (
    <>
      <EscMenuContainer isOpen={isOpen} onClose={onClose}></EscMenuContainer>
      <Box
        fontSize="xs"
        className="toolbar"
        padding={2}
        display="flex"
        alignItems="center"
        justifyContent="space-between"
        bgColor={color}
        borderColor={borderColor}
        borderBottomStyle="solid"
        borderBottomWidth="1px"
        {...responsiveFontProps}
      >
        <HStack>
          <Button
            onClick={() => {
              layoutVar({ sideNav: !sideNav, toolBar });
            }}
            {...responsiveFontProps}
          >
            <ChatIcon {...responsiveFontProps}></ChatIcon>
          </Button>
          <Button onClick={() => onOpen()} {...responsiveFontProps}>
            <SettingsIcon {...responsiveFontProps}></SettingsIcon>&nbsp;{' '}
            <Kbd>Esc</Kbd>
          </Button>
        </HStack>
        <Auth></Auth>
      </Box>
    </>
  );
};
