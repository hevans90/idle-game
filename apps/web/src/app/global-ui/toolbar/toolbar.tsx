import { useReactiveVar } from '@apollo/client';
import { ChatIcon, SettingsIcon } from '@chakra-ui/icons';
import {
  Box,
  Button,
  HStack,
  Icon,
  Kbd,
  useDisclosure,
} from '@chakra-ui/react';
import { BsFileMusic } from 'react-icons/bs';

import { Auth } from '../../_auth/auth';
import {
  responsiveFontProps,
  responsiveIconProps,
} from '../../_responsive-utils/font-props';

import {
  globalUiVar,
  hotkeyHintsVar,
  layoutVar,
  musicPlayerVar,
} from '@idleverse/state';
import { useUiBackground } from '@idleverse/theme';
import { useKeypress } from '../../hooks/use-keypress';
import { EscMenuContainer } from '../esc-menu/escape-menu.container';

export const ToolBar = () => {
  const { bg, border } = useUiBackground();

  useKeypress('Escape', () => {
    if (!escapeMenuOpen) {
      globalUiVar({ ...globalUiVar(), escapeMenuOpen: true });
    } else {
      globalUiVar({ ...globalUiVar(), escapeMenuOpen: false });
    }
  });

  const { onClose } = useDisclosure();

  const { sideNav, toolBar } = useReactiveVar(layoutVar);

  const { escapeMenuOpen } = useReactiveVar(globalUiVar);
  const { show: showMusicPlayer } = useReactiveVar(musicPlayerVar);

  const hotkeyHints = useReactiveVar(hotkeyHintsVar);

  return (
    <>
      <EscMenuContainer
        isOpen={escapeMenuOpen}
        onClose={() => {
          globalUiVar({ ...globalUiVar(), escapeMenuOpen: false });
          onClose();
        }}
      ></EscMenuContainer>
      <Box
        fontSize="xs"
        className="toolbar"
        padding={2}
        display="flex"
        alignItems="center"
        justifyContent="space-between"
        bgColor={bg}
        borderColor={border}
        borderBottomStyle="solid"
        borderBottomWidth="1px"
        {...responsiveFontProps}
      >
        <HStack>
          <Button
            onClick={() => {
              layoutVar({ sideNav: !sideNav, toolBar });
            }}
            {...responsiveFontProps}
          >
            <ChatIcon {...responsiveIconProps}></ChatIcon>
          </Button>
          <Button
            onClick={() =>
              globalUiVar({ ...globalUiVar(), escapeMenuOpen: !escapeMenuOpen })
            }
            {...responsiveFontProps}
          >
            <SettingsIcon {...responsiveIconProps}></SettingsIcon>
            {hotkeyHints && (
              <>
                &nbsp; <Kbd>Esc</Kbd>
              </>
            )}
          </Button>
          <Button
            onClick={() => {
              musicPlayerVar({ ...musicPlayerVar(), show: !showMusicPlayer });
            }}
            {...responsiveFontProps}
          >
            <Icon as={BsFileMusic} {...responsiveIconProps} />
          </Button>
        </HStack>
        <Auth></Auth>
      </Box>
    </>
  );
};
