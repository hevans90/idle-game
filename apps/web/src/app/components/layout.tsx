import { useReactiveVar } from '@apollo/client';
import { Box } from '@chakra-ui/layout';
import { useBreakpointValue } from '@chakra-ui/react';
import styled from '@emotion/styled';

import { LayoutConfig } from '@idleverse/models';
import { layoutVar } from '@idleverse/state';
import { useUiBackground } from '@idleverse/theme';
import { SideNav } from '../global-ui/sidenav/sidenav';
import { ToolBar } from '../global-ui/toolbar/toolbar';
import { Breadcrumb } from './breadcrumb';

export const sideNavWidth = 450;
export const topBarHeight = 50;

export const ResponsiveGrid = styled.div`
  height: 100vh;

  display: grid;

  grid-template-rows: ${topBarHeight}px auto;

  grid-template-areas: ${({ sideNav }: LayoutConfig) =>
    sideNav
      ? `'toolbar'
         'side-nav'
         `
      : `'toolbar'
        ' main   '`};

  @media (min-width: 768px) {
    grid-template-areas: ${({ sideNav }: LayoutConfig) =>
      sideNav
        ? `'side-nav   toolbar'
           'side-nav   main   '
         `
        : `'toolbar'
        ' main   '`};

    grid-template-columns: ${({ sideNav }: LayoutConfig) =>
      sideNav ? `${sideNavWidth}px auto` : `auto`};
  }

  div.sidenav {
    grid-area: side-nav;
  }
  div.toolbar {
    grid-area: toolbar;
  }

  main {
    grid-area: main;
    position: relative;

    .container {
      height: calc(100vh - ${topBarHeight}px);
    }
  }
`;

export const Layout = (props: { children?: JSX.Element }) => {
  const layoutConfig = useReactiveVar(layoutVar);

  const { bgDark } = useUiBackground();

  const bp: 'small' | 'medium' | 'large' = useBreakpointValue({
    base: 'small',
    md: 'medium',
    lg: 'large',
  });

  return (
    <ResponsiveGrid {...layoutConfig}>
      {layoutConfig.sideNav ? <SideNav></SideNav> : null}
      <ToolBar></ToolBar>
      {((bp === 'small' && !layoutConfig.sideNav) || bp !== 'small') && (
        <main>
          <Breadcrumb />
          <Box className="container" overflow="auto" bg={bgDark}>
            {props?.children}
          </Box>
        </main>
      )}
    </ResponsiveGrid>
  );
};
