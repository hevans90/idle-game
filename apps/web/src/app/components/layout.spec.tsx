import { ToolBar } from '../containers/toolbar/toolbar';
import { renderWithChakra } from '../_test-utils/render-with-chakra';
import { ResponsiveGrid } from './layout';

jest.mock('../containers/toolbar/toolbar', () => ({
  ToolBar: () => <div className="toolbar"></div>,
}));

describe('layout', () => {
  it('should render', () => {
    const { baseElement } = renderWithChakra(
      <ResponsiveGrid sideNav={false} toolBar={false}>
        <ToolBar></ToolBar>
      </ResponsiveGrid>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
