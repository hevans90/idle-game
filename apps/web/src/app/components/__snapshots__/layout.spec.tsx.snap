// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout should render 1`] = `
<body
  class="chakra-ui-dark"
>
  <div>
    <div
      class="css-rsr9j6"
    >
      <div
        class="toolbar"
      />
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>
  <div
    class="chakra-portal"
  >
    <div
      aria-label="Notifications"
      aria-live="polite"
      id="chakra-toast-manager-top"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <div
      aria-label="Notifications"
      aria-live="polite"
      id="chakra-toast-manager-top-left"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
    />
    <div
      aria-label="Notifications"
      aria-live="polite"
      id="chakra-toast-manager-top-right"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
    />
    <div
      aria-label="Notifications"
      aria-live="polite"
      id="chakra-toast-manager-bottom-left"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
    />
    <div
      aria-label="Notifications"
      aria-live="polite"
      id="chakra-toast-manager-bottom"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <div
      aria-label="Notifications"
      aria-live="polite"
      id="chakra-toast-manager-bottom-right"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
    />
  </div>
</body>
`;
